src/scripts/commands/database.ts

```diff
@@ -0,0 +1,205 @@
+    const removeOrgUnitLevelCommand = command({
+        name: "remove-orgunit-level",
+        description: "Create a sql file that removes organisation units below a hierarchy level",
+        args: {
+            level: positional({
+                type: HierarchyLevel,
+                description: "Maximum hierarchy level to keep",
+            }),
+            outputFile: option({
+                type: optional(string),
+                long: "output-file",
+                description: "Name of the output file that will contain the sql commands",
+            }),
+            overwrite: flag({
+                long: "overwrite",
+                description: "Overwrite ouput file if exists",
+            }),
+        },
+        handler: async args => {
+            const outputFile = args.outputFile ?? "remove_orgunits.sql";
+            const sqlCommands = removeOrgs(`hierarchylevel > ${args.level}`);
+            safeWrite(outputFile, sqlCommands, args.overwrite);
+            process.exit(0);
+        },
+    });
+
+    const removeOrgUnitChildrenCommand = command({
+        name: "remove-orgunit-children",
+        description: "Create a sql file that removes all children of an organisation unit",
+        args: {
+            path: positional({
+                type: OrgUnitPath,
+                description: "Path of the organisation unit to prune of its children",
+            }),
+            outputFile: option({
+                type: optional(string),
+                long: "output-file",
+                description: "Name of the output file that will contain the sql commands",
+            }),
+            overwrite: flag({
+                long: "overwrite",
+                description: "Overwrite ouput file if exists",
+            }),
+        },
+        handler: async args => {
+            const outputFile = args.outputFile ?? "remove_orgunits.sql";
+            const sqlCommands = removeOrgs(`path LIKE '${args.path}/%'`);
+            safeWrite(outputFile, sqlCommands, args.overwrite);
+            process.exit(0);
+        },
+    });
+
+    return subcommands({
+        name: "database",
+        cmds: {
+            "remove-orgunit-level": removeOrgUnitLevelCommand,
+            "remove-orgunit-children": removeOrgUnitChildrenCommand,
+        },
+    });
+}
+
+// Write to outputFile the given string str, unless the file exists.
+function safeWrite(outputFile: string, str: string, overwrite: boolean) {
+    if (!fs.existsSync(outputFile) || overwrite) {
+        fs.writeFileSync(outputFile, str);
+        console.log(`Written: ${outputFile}`);
+    } else {
+        console.error(`ERROR. Output file already exists: ${outputFile}`);
+        console.error("Use --overwrite if you want to overwrite it anyway.");
+    }
+}
+
+// Return an SQL string that would remove all the organisation units that
+// satisfy the given whereCondition.
+function removeOrgs(whereCondition: string): string {
```

Comment:
Following clean architecture guidelines, from `safeWrite` to the end should not live here. I'd create (assuming we only have one command with options):

- Use case: src/domain/usecases/CreateSQLForOrgUnitRemovalUseCase.ts 
- Repositories: interface -> src/domain/repositories/OrgUnitsRepository.ts (getDeleteScript) + src/data/OrgUnitD2Repository.ts (with the implementation)

Reply:
I've been trying for quite a while to wrap my head around this one. I don't get it. This is the only thing I have not addressed. I think I would need help with this.

Reply:
You can ask Nacho how to tackle this, maybe Jorge can step in. The code is very located and won't hurt a non-clean-arch approach for now, but in the long run, we should keep all the package on clean-arch.

Reply:
@jordibc @tokland I think Arnau's idea is very good. Jordi, could you ask Jorge to check this PR and contribute with the changes that would make this clean? If you could afterwards have a short meeting with him so he can explain it to you, that would be great. Thank you!

Reply:
Hi @jordibc, @ifoche asked me to review this PR about clean architecture.

I agree with @tokland, the code in the handle should be moved to a use case.

The goal of clean architecture is separate responsibilities and hide details of implementation. 

We are going to review problems with the current implementations.

* The logic under org units command is coupled to cmd-ts
* The infrastructure logic to read and write files is coupled to fs
* Here there are three responsibilities:
   * Presentation -> cmd-ts related code
   * Domain -> The coordination of the actions to realize:
       1. validations
       2. Create where condition
       3. get the template
       4. Create sqlCommands 
       5. Save sqlCommands
       6. Return sqlCommands
   * Infrastructure o data -> read the template from disk and save the sqlCommands to disk

But all three responsibilities are in the same file in a coupled approach, consequences:

* Homogeneity -> other commands in this project follow clean architecture less this. This creates a problem, for example, the next person will have doubts about when to use cases o not.
* You are the code creator but for example, if another person that doesn't know the code modifies it, how are the 3 responsibilities together maybe wanting to change one creates a bug in another. The classic problem of fixing one thing and breaking another.
* You can not create an integration test that executes the logic in the org units command without using cmd-ts and verifying the final output is the expected
* You can not create unit tests that verify that expected validations are realized

How to migrate to clean architecture:
- Presentation: read args, invoke to the use case, the exit of the process when the use case returns errors o throws exceptions, the exit of the process when to the finish
- Domain: create a use case with the coordination tasks
- Infrastructure or data: create a repository to read and write files on disk

I think that reviewing code in other commands in this same project can help.